### PR TITLE
Better change detection

### PR DESCRIPTION
**Pull request by kflynn**
Do a better job of handling corner cases in detecting changes in mappings.

### CODE REVIEW

File: ambassador/ambassador.py [hunk 1]
```diff
@@ -295,18 +295,37 @@ def ambassador_stats():
...
+        app.current_mappings = None
+
+        logging.debug("new_config got no mappings at all? %s" % str(rc))
+        return RichStatus.fromError("no mappings found at all, original rc %s" % str(rc))
+
+    num_mappings = 0
+
+    try:
+        num_mappings = len(rc.mappings)
+    except:
```
Comment (ark3) [inline ambassador/ambassador.py]: if `except TypeError:` doesn't do the job here, I would be very surprised.

Comment (kflynn) [inline ambassador/ambassador.py]: I thought about that, and decided to deliberately go broad. 
Open to persuasion about why that would be a bad idea.

File: ambassador/ambassador.py [hunk 2]
```diff
@@ -295,18 +295,37 @@ def ambassador_stats():
...
+        # Huh. This can really only happen if something isn't quite initialized
+        # in the database yet.
+        logging.debug("new_config mappings is not an array, assuming empty")
```
Comment (ark3) [inline ambassador/ambassador.py]: Weird case. May as well log the `type(...)` of mappings.

File: ambassador/ambassador.py (hunk 1 above)
Comment (ark3) [inline ambassador/ambassador.py]: All the usual reasons why `except:` is usually wrong (as compared to `except Exception:` below), but I doubt this is at all important.

**Pull request merged**